mongodbatlas/resource_mongodbatlas_org_invitation_test.go

```diff
@@ -0,0 +1,198 @@
+		orgID        = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		name         = fmt.Sprintf("test-acc-%s", acctest.RandString(10))
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckMongoDBAtlasOrgInvitationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasOrgInvitationConfig(orgID, name, []string{"mongodbatlas.testing@gmail.com"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "org_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "name"),
+					resource.TestCheckResourceAttrSet(resourceName, "usernames.#"),
+					resource.TestCheckResourceAttrSet(resourceName, "invitation_id"),
+
+					resource.TestCheckResourceAttr(resourceName, "org_id", orgID),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "usernames.#", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportStateIdFunc: testAccCheckMongoDBAtlasOrgInvitationStateIDFunc(resourceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckMongoDBAtlasOrgInvitationExists(resourceName string, invitation *matlas.Invitation) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*MongoDBClient).Atlas
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		ids := decodeStateID(rs.Primary.ID)
+
+		orgID := ids["org_id"]
+		username := ids["username"]
+		invitationID := ids["invitation_id"]
+
+		if orgID == "" && username == "" && invitationID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		log.Printf("[DEBUG] orgID: %s", orgID)
+		log.Printf("[DEBUG] username: %s", username)
+		log.Printf("[DEBUG] invitationID: %s", invitationID)
+
+		invitationResp, _, err := conn.Organizations.Invitation(context.Background(), orgID, invitationID)
+		if err == nil {
+			*invitation = *invitationResp
+			return nil
+		}
+
+		return fmt.Errorf("invitation(%s) does not exist", invitationID)
+	}
+}
+
+func testAccCheckMongoDBAtlasOrgInvitationUsernameAttribute(invitation *matlas.Invitation, username string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if invitation.Username != username {
+			return fmt.Errorf("bad name: %s", invitation.Username)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckMongoDBAtlasOrgInvitationRoleAttribute(invitation *matlas.Invitation, roles []string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if len(roles) > 0 {
+			for _, role := range roles {
+				for _, currentRole := range invitation.Roles {
+					if currentRole == role {
+						return nil
+					}
+				}
+			}
+		}
+
+		return fmt.Errorf("bad role: %s", invitation.Roles)
```

Comment:
[nit] it should be safe to skip the length check 
```suggestion
		for _, role := range roles {
			for _, currentRole := range invitation.Roles {
				if currentRole == role {
					return nil
				}
			}
		}
		return fmt.Errorf("bad role: %s", invitation.Roles)
```

Reply:
Done, thanks